Deepak_Yadav/p_website/WebApplication2/WebApplication2/controllers/companyinfocontroller.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+
+// For more information on enabling Web API for empty projects, visit https://go.microsoft.com/fwlink/?LinkID=397860
+
+namespace WebApplication2.controllers
+{
+    [Route("api/[controller]")]
+    public class companyinfocontroller : Controller
+    {
+        Models.companyinfoContext _context;
+        public companyinfocontroller(Models.companyinfoContext context)
+        {
+            _context = context;
+        }
+
+        // GET: api/<controller>
+        [HttpGet]
+        public ActionResult getStudent([FromBody]Custom_Models.tcompanycustom data)
+        {
+            //DbModels.StudentInfo student = new DbModels.StudentInfo();
+            List<Models.Tcompanydata> employees = _context.Tcompanydata.ToList();
+            return Ok(employees);
+        }
+
+        // GET api/<controller>/5
+        [HttpGet("{id}")]
+        public string Get(int id)
+        {
+            return "value";
+        }
+
+        // POST api/<controller>
+        [HttpPost]
+        public ActionResult Post([FromBody]Custom_Models.tcompanycustom data)
+        {
+            Models.Tcompanydata emp = new Models.Tcompanydata()
+            {
+           
+            Name = data.Name,
+            PhoneNo = data.PhoneNo,
+            Designation = data.Designation,
+            Email = data.Email,
+            Password=data.Password,
+        };
+            _context.Tcompanydata.Add(emp);
+            _context.SaveChanges();
+            return Ok("employees added");
+        }
+
+        // PUT api/<controller>/5
+        [HttpPut("{id}")]
+        public void Put(int id, [FromBody]string value)
+        {
+        }
+
+        [HttpDelete]
+        public ActionResult deleteStudent([FromBody]Custom_Models.tcompanycustom data)
```

Comment:
Delete should be based on Id. _data_ parameter is not required as well as not being used anywhere in this method. This code would delete the first entry from employees table.

Reply:
Please Do not resolve any Comment yourself .